vcd/resource_vcd_vapp_vm.go

```diff
@@ -853,6 +883,17 @@ func resourceVcdVAppVmRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("href", vm.VM.HREF)
 	d.Set("expose_hardware_virtualization", vm.VM.NestedHypervisorEnabled)
 
+	// update guest properties
+	guestProperties, err := vm.GetGuestProperties()
+	if err != nil {
+		return fmt.Errorf("unable to read guest properties: %s", err)
+	}
+
+	err = setProductSectionListData(d, guestProperties)
```

Comment:
When looking at govcd PR, it looked like the idea was to hide this "product section" naming from the caller and expose "guest properties" instead?

Reply:
These functions worked in the "concept" of guest properties, but the API calls are ProductSection. I'll look if I can refine them somehow clearer

Reply:
I think here we operate with State file, in state file we have guest properties.

Reply:
We agreed to use `setXData` and `getXType` where X=struct name

Reply:
@dataclouder your thoughts?

Reply:
> We agreed to use `setXData` and `getXType` where X=struct name

Yes, but I think set/GetTerrraformStructure, IMHO. In govcd API struct.

Reply:
I have changed govcd method namings